apps/query-ui/queryui/main.py

```diff
@@ -0,0 +1,85 @@
+#!/usr/bin/env python
+
+# This script is used to run the Streamlit-based Luna Query UI.
+# Because of issues with Ray, we need to initialize Ray externally
+# to the demo process, and then run the Streamlit UI in a subprocess.
+
+import argparse
+import logging
+import os
+import subprocess
+import time
+
+import ray
+
+from sycamore.executor import _ray_logging_setup
+
+
+def ray_init(**ray_args):
+    """Used to initialize Ray before running the Streamlit app."""
+    if ray.is_initialized():
+        return
+
+    if "logging_level" not in ray_args:
+        ray_args.update({"logging_level": logging.INFO})
+    if "runtime_env" not in ray_args:
+        ray_args["runtime_env"] = {}
+    if "worker_process_setup_hook" not in ray_args["runtime_env"]:
+        ray_args["runtime_env"]["worker_process_setup_hook"] = _ray_logging_setup
+    ray.init(**ray_args)
+
+    while not ray.is_initialized():
+        print("Waiting for ray to initialize...", flush=True)
+        time.sleep(1)
+
+
+def main():
+    argparser = argparse.ArgumentParser()
+    argparser.add_argument("--chat", action="store_true", help="Only show the chat demo pane.")
+    argparser.add_argument(
+        "--index", help="OpenSearch index name to use. If specified, only this index will be queried."
+    )
+    argparser.add_argument(
+        "--llm-cache-dir", type=str, default="llm_cache", help="LLM query cache dir. Defaults to ./llm_cache."
```

Comment:
probably want to use cache-dir from https://github.com/aryn-ai/sycamore/pull/850/files#diff-d6138cc332b0d2bb7e774f837ecceb3056c656c38b1faafff1b1703fbdb6b340 ?

Reply:
Yes, done. I had sent the PR for that concurrently to this one :-)